pkg/trace/api/otlp.go

```diff
@@ -363,39 +357,85 @@ func convertSpan(rattr map[string]string, lib pdata.InstrumentationLibrary, in p
 	if in.Events().Len() > 0 {
 		span.Meta["events"] = marshalEvents(in.Events())
 	}
+	var ctags strings.Builder // collect container tags from attributes
 	in.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
 		switch v.Type() {
 		case pdata.AttributeValueTypeDouble:
 			span.Metrics[k] = v.DoubleVal()
 		case pdata.AttributeValueTypeInt:
 			span.Metrics[k] = float64(v.IntVal())
 		default:
-			span.Meta[k] = v.AsString()
+			switch k {
+			case "operation.name":
+				span.Name = v.AsString()
+			case "service.name":
+				span.Service = v.AsString()
+			case "resource.name":
+				span.Resource = v.AsString()
+			case "span.type":
+				span.Type = v.AsString()
+			default:
+				span.Meta[k] = v.AsString()
+			}
+		}
+		if containerTagsAttributes[k] {
+			// attribute matching container tag found
+			if ctags.Len() > 0 {
+				ctags.WriteByte(',')
+			}
+			ctags.WriteString(conventionsMapping[k])
+			ctags.WriteByte(':')
+			ctags.WriteString(v.AsString())
 		}
 		return true
 	})
+	if ctags.Len() > 0 {
+		span.Meta[tagContainersTags] = ctags.String()
+	}
 	if _, ok := span.Meta["env"]; !ok {
 		if env := span.Meta[string(semconv.AttributeDeploymentEnvironment)]; env != "" {
-			span.Meta["env"] = env
+			span.Meta["env"] = traceutil.NormalizeTag(env)
 		}
 	}
 	if in.TraceState() != pdata.TraceStateEmpty {
 		span.Meta["trace_state"] = string(in.TraceState())
 	}
 	if lib.Name() != "" {
-		span.Meta["instrumentation_library.name"] = lib.Name()
+		span.Meta["otel.library.name"] = lib.Name()
```

Comment:
Use these instead https://pkg.go.dev/go.opentelemetry.io/collector/model@v0.48.0/semconv/v1.9.0#pkg-constants, here and below. Also, these changed name recently (see [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/non-otlp.md#instrumentationscope)) so this will be out of date soon.